src/DataTypes/Serializations/SerializationNumber.cpp

```diff
@@ -142,7 +142,21 @@ void SerializationNumber<T>::serializeBinaryBulk(const IColumn & column, WriteBu
         limit = size - offset;
 
     if (limit)
-        ostr.write(reinterpret_cast<const char *>(&x[offset]), sizeof(typename ColumnVector<T>::ValueType) * limit);
+    {
+        if constexpr (std::endian::native == std::endian::big && sizeof(T) >= 2)
+        {
+            for (size_t i = 0; i < limit; i++)
+            {
+                auto tmp(x[offset+i]);
+                char *start = reinterpret_cast<char*>(&tmp);
+                char *end = start + sizeof(typename ColumnVector<T>::ValueType);
+                std::reverse(start, end);
```

Comment:
`transformEndianness` should help you out here as well.

Reply:
Reverted this change based on latest code.

Reply:
Reverted this change based on latest code.